### PR TITLE
[Linux] Publish Specialization complete event from functions host

### PR DESCRIPTION
**Pull request by balag0**
This event will be used by the init process in the container to track functions runtime status. A delay of 5 seconds is added to avoid impacting cold start.

https://github.com/Azure/azure-functions-docker-private/issues/136

resolves https://github.com/Azure/azure-functions-docker-private/issues/136

### Pull request checklist

* [X ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ X] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR https://github.com/Azure/azure-functions-docker-private/issues/136
* [ X] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs [hunk 1]
```diff
@@ -16,23 +16,27 @@ namespace Microsoft.Azure.WebJobs.Script.WebHost.ContainerManagement
 {
     public class LinuxContainerActivityPublisher : IHostedService, IDisposable, ILinuxContainerActivityPublisher
     {
+        public const string SpecializationCompleteEvent = "Specialization complete";
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs]: nit: move this to `ScriptContants` and update the value to be more descriptive:
`public const string LinuxConsumptionSpecializationCompletedEvent = "Specialized";` ?

File: src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs [hunk 2]
```diff
@@ -96,8 +102,30 @@ public Task StopAsync(CancellationToken cancellationToken)
...
+        private async Task PublishSpecializationCompleteEvent()
+        {
+            try
+            {
+                await _meshServiceClient.NotifyHealthEvent(ContainerHealthEventType.Informational, GetType(),
+                    SpecializationCompleteEvent);
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"{nameof(PublishSpecializationCompleteEvent)} failed with {e}");
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs]: nit: use parameterized logging 
` _logger.LogWarning("{PublishSpecializationCompleteEvent} failed with {ex}", nameof(PublishSpecializationCompleteEvent), ex);`

**Review by pragnagopa — COMMENTED** [review]
Added minor comments. Thanks!

File: src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs (hunk 2 above)
Comment (balag0) [reply src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs]: Fixed.

File: src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs (hunk 1 above)
Comment (balag0) [reply src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs]: Yes, I am thinking this is not needed outside this class (other than the test) so leaning towards leaving it here for now and move it to global constants if things change later?

Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs]: Ok to leave it here. I recommend changing it to `Specialized` or `SpecializationCompleted` - without the space

**Review by pragnagopa — DISMISSED** [review]

File: src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs (hunk 1 above)
Comment (balag0) [reply src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerActivityPublisher.cs]: Yes. Fixed in the current iteration.

**Review by pragnagopa — APPROVED** [review]

**Pull request merged**